runtime/src/bank.rs

```diff
@@ -363,55 +376,39 @@ impl AbiExample for CachedExecutors {
     }
 }
 
-impl Clone for CachedExecutors {
-    fn clone(&self) -> Self {
-        let executors = self.executors.iter().map(|(&key, entry)| {
-            let entry = CachedExecutorsEntry {
-                prev_epoch_count: entry.prev_epoch_count,
-                epoch_count: AtomicU64::new(entry.epoch_count.load(Relaxed)),
-                executor: entry.executor.clone(),
-                hit_count: AtomicU64::new(entry.hit_count.load(Relaxed)),
-            };
-            (key, entry)
-        });
+impl CachedExecutors {
+    fn new(max_capacity: usize, current_epoch: Epoch) -> Self {
         Self {
-            max: self.max,
-            current_epoch: self.current_epoch,
-            executors: executors.collect(),
+            max_capacity,
+            current_epoch,
+            executors: HashMap::new(),
             stats: executor_cache::Stats::default(),
         }
     }
-}
 
-impl CachedExecutors {
-    fn clone_with_epoch(self: &Arc<Self>, epoch: Epoch) -> Arc<Self> {
-        if self.current_epoch == epoch {
-            return self.clone();
-        }
-        let executors = self.executors.iter().map(|(&key, entry)| {
-            // The total_count = prev_epoch_count + epoch_count will be used for LFU eviction.
-            // If the epoch has changed, we store the prev_epoch_count and reset the epoch_count to 0.
-            let entry = CachedExecutorsEntry {
-                prev_epoch_count: entry.epoch_count.load(Relaxed),
-                epoch_count: AtomicU64::default(),
-                executor: entry.executor.clone(),
-                hit_count: AtomicU64::new(entry.hit_count.load(Relaxed)),
-            };
-            (key, entry)
-        });
-        Arc::new(Self {
-            max: self.max,
-            current_epoch: epoch,
-            executors: executors.collect(),
-            stats: executor_cache::Stats::default(),
-        })
-    }
+    fn new_from_parent(parent: &CachedExecutors, current_epoch: Epoch) -> Self {
```

Comment:
This `..._from_parent` is kind of odd because a cache doesn't really have a parent.
A `with_epoch` might be better. Something like

```rust
fn with_epoch(self, epoch: Epoch) -> Self {
   if self.current_epoch == epoch {
       return self;
   }
   // change current_epoch, reset counters, etc 
   self
}
```

so then you can call
```rust
cache.clone().with_epoch(epoch)
```

Reply:
Opted to use `new_from_parent_bank_executors` to be more explicit